omod/src/main/java/org/openmrs/module/appointments/web/controller/RecurringAppointmentsController.java

```diff
@@ -160,4 +164,38 @@ public RecurringAppointmentDefaultResponse getAppointmentByUuid(@RequestParam(va
         }
         return recurringAppointmentMapper.constructResponse(appointment);
     }
+
+    @RequestMapping(method = RequestMethod.POST, value = "/conflicts")
+    @ResponseBody
+    public ResponseEntity<Object> conflicts(@RequestBody RecurringAppointmentRequest recurringAppointmentRequest) {
+        try {
+            RecurringPattern recurringPattern = recurringAppointmentRequest.getRecurringPattern();
```

Comment:
Extract this to private method and use proper error codes. Add timezone validator

Reply:
Done

Reply:
Extract private method and timezone should be only in edit flow not in save flow

Reply:
yes done.